pytest/array_view_test.py

```diff
@@ -0,0 +1,198 @@
+from itertools import product
+
+import numpy as np
+
+import arkouda as ak
+import pytest
+
+
+def test_mulitdimensional_array_creation():
+    n = np.array([[0, 0], [0, 1], [1, 1]])
+    a = ak.array([[0, 0], [0, 1], [1, 1]])
+    assert (n.tolist() == a.to_list())
+    n = np.arange(27).reshape((3, 3, 3))
+    a = ak.arange(27).reshape((3, 3, 3))
+    assert (n.tolist() == a.to_list())
+    n = np.arange(27).reshape(3, 3, 3)
+    a = ak.arange(27).reshape(3, 3, 3)
+    assert (n.tolist() == a.to_list())
+    n = np.arange(27, dtype=np.uint64).reshape(3, 3, 3)
+    a = ak.arange(27, dtype=ak.uint64).reshape(3, 3, 3)
+    assert (n.tolist() == a.to_list())
+
+
+def test_arrayview_int_indexing():
+    nd = np.arange(9).reshape(3, 3)
+    pd_reshape = ak.arange(9).reshape(3, 3)
+    pd_array = ak.array([[0, 1, 2], [3, 4, 5], [6, 7, 8]])
+
+    nd_ind = [nd[i, j] for (i, j) in product(range(3), repeat=2)]
+    reshape_ind = [pd_reshape[i, j] for (i, j) in product(range(3), repeat=2)]
+    array_ind = [pd_array[i, j] for (i, j) in product(range(3), repeat=2)]
+    assert (nd_ind == reshape_ind)
+    assert (nd_ind == array_ind)
+
+    with pytest.raises(IndexError):
+        # index out bounds (>= dimension)
+        # index 3 is out of bounds for axis 0 with size 3
+        pd_reshape[3, 1]

```

Comment:
Just for clarity and documentation of the changes, can you provide an example of the `pytest.raises` failing compared to the old test? Based on the `assert` functionality, I don't see it being an issue, but it would be good to document moving forward.

Reply:
Since this is asserting that the error type was raised, on fails it really just says that the expected error wasn't raised
```
        with pytest.raises(IndexError):
            # index out bounds (>= dimension)
            # index 3 is out of bounds for axis 0 with size 3
>           pd_reshape[1, 1]
E           Failed: DID NOT RAISE <class 'IndexError'>
```

The previous implementation:
```
        with self.assertRaises(IndexError):
            # index out bounds (>= dimension)
            # index 3 is out of bounds for axis 0 with size 3
>           pd_reshape[1, 1]
E           AssertionError: IndexError not raised
```